### DISCUSSION TITLE
How understand value about size during enabled dedup and compression on zpool volume

### DISCUSSION THREAD

**Discussion by Neptunet-GitHub** [openzfs/zfs#16783] [Q&A] [answered]
Hi Guys,

I created RAIDZ3 based on 9 disks 3,7 TB each. It created zfs device with file system with 20 TB usage space. Next I enabled deduplication and compression on it + filesystem is synthetic. Everything is correct.

Now I wrote files on the file system.

And from this time I lost understanding of reporting information.

When I try to check how much free space I have from operating system I got:
![image](https://github.com/user-attachments/assets/6379ca39-9957-46c6-853b-4f1520552ece)

Next I checked ZFS statistics:
![image](https://github.com/user-attachments/assets/e6ba6167-f4ce-42e3-acef-33fb12b8462c)

From Linux reports, that file system have size 106T and free 18T,
From ZFS reports. that file system have size 31,4T and free 27,9T, also ALLOCATION is 3,55T,
but
ZFS reports also that used 88,4T and available 17T.

Is any document or faq which explain how to understand these values? Sorry, but I cannot found it. 

for example, How much free space I have on that file system? Real information is from ZFS - that allocation is 3,55T and I had 20TB total space on empty ZFS FS, so I have about 16,4T free?
Why ZFS reports that free is 27,9T and size is 31.4T, as size was 20T? Operating system reports that free is 18T, but it also isn't true, because I wrote on it additional 100TB and allocation before it was 3,52T after wrote that 100TB allocation is 3,55T

I try to understand how count these values, but I have little problem with that.

**Comment by amotin** [comment, marked as answer]
When you are looking on pool stats, it shows raw space (including parity) for RAIDZ, that is why there was 31.3TiB vs 20TiB originally reported free/available.  Another factor here is huge dedup ratio you've got, that you may see in the pool stats.  Since dedup does not work on dataset level, ZFS can not report it there, so it inflates pool capacity, as if all the data you have written are really stored somewhere.  The 18TiB of space reported as available for dataset should be right, if you finally start writing real unique data.  And it matches FREE reported for the pool if you compensate for parity.
